public/app/features/explore/extensions/ToolbarExtensionPoint.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { lazy, ReactElement, Suspense, useMemo, useState } from 'react';
+
+import { type PluginExtensionLink, PluginExtensionPoints, RawTimeRange } from '@grafana/data';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { DataQuery, TimeZone } from '@grafana/schema';
+import { Dropdown, Menu, ToolbarButton } from '@grafana/ui';
+import { contextSrv } from 'app/core/services/context_srv';
+import { truncateTitle } from 'app/features/plugins/extensions/utils';
+import { AccessControlAction, ExploreId, ExplorePanelData, useSelector } from 'app/types';
+
+import { getExploreItemSelector } from '../state/selectors';
+
+import { ConfirmNavigationModal } from './ConfirmNavigationModal';
+
+const AddToDashboard = lazy(() =>
+  import('./AddToDashboard').then(({ AddToDashboard }) => ({ default: AddToDashboard }))
+);
+
+type Props = {
+  exploreId: ExploreId;
+  timeZone: TimeZone;
+  splitted: boolean;
+};
+
+export function ToolbarExtensionPoint(props: Props): ReactElement | null {
+  const { exploreId, splitted } = props;
+  const [extension, setExtension] = useState<PluginExtensionLink | undefined>();
```

Comment:
Minor: maybe `selectedExtension` instead?
```suggestion
  const [selectedExtension, setSelectedExtension] = useState<PluginExtensionLink | undefined>();
```

Reply:
Sounds good! Will commit 👍🏻